x-pack/plugins/lens/public/app_plugin/mounter.tsx

```diff
@@ -105,6 +116,7 @@ export async function mountApp(
       <HashRouter>
         <Switch>
           <Route exact path="/edit/:id" render={renderEditor} />
+          <Route exact path={`/${LENS_EDIT_BY_VALUE}`} render={renderEditor} />
```

Comment:
I have to think that there is some improvement that we could make to this API so that the URL can be used to simplify the code. The amount of `state.byValueMode` logic that you've added in this PR suggests worries me.